Puppetfile.lock

```diff
@@ -19,141 +19,105 @@ FORGE
       puppetlabs-concat (< 5.0.0, >= 1.0.0)
       puppetlabs-stdlib (< 5.0.0, >= 4.13.0)
-    theforeman-dns (5.0.1)
+    theforeman-dns (5.1.0)
       puppetlabs-concat (< 5.0.0, >= 1.0.0)
       puppetlabs-stdlib (< 5.0.0, >= 4.13.1)
-    theforeman-foreman (8.1.1)
+    theforeman-foreman (9.0.0)
       puppet-extlib (< 3.0.0, >= 0.10.4)
       puppetlabs-apache (< 3.0.0, >= 2.0.0)
       puppetlabs-apt (< 5.0.0, >= 2.0.0)
       puppetlabs-concat (< 5.0.0, >= 1.0.0)
       puppetlabs-postgresql (< 6.0.0, >= 4.2.0)
       puppetlabs-stdlib (< 5.0.0, >= 4.13.0)
-      theforeman-tftp (< 4.0.0, >= 3.0.0)
-    theforeman-foreman_proxy (6.0.3)
+    theforeman-foreman_proxy (7.0.0)
       puppet-extlib (< 3.0.0, >= 0.10.4)
       puppetlabs-stdlib (< 5.0.0, >= 4.19.0)
       richardc-datacat (< 1.0.0, >= 0.6.0)
       theforeman-dhcp (< 5.0.0, >= 1.3.0)
       theforeman-dns (< 6.0.0, >= 1.3.0)
-      theforeman-foreman (< 9.0.0, >= 8.0.0)
+      theforeman-foreman (< 10.0.0, >= 8.0.0)
       theforeman-puppet (< 9.0.0, >= 4.2.0)
-      theforeman-tftp (< 4.0.0, >= 1.3.0)
-    theforeman-git (3.0.0)
-    theforeman-puppet (8.0.4)
+      theforeman-tftp (< 5.0.0, >= 1.3.0)
+    theforeman-puppet (8.2.0)
       puppet-extlib (< 3.0.0, >= 0.11.3)
       puppetlabs-apache (< 3.0.0, >= 1.2.0)
       puppetlabs-concat (< 5.0.0, >= 1.0.0)
+      puppetlabs-hocon (< 2.0.0, >= 1.0.0)
+      puppetlabs-puppet_authorization (< 1.0.0, >= 0.3.0)
       puppetlabs-stdlib (< 5.0.0, >= 4.13.0)
-    theforeman-tftp (3.0.2)
+    theforeman-tftp (4.0.0)
       puppetlabs-stdlib (< 5.0.0, >= 4.13.1)
       puppetlabs-xinetd (< 4.0.0, >= 1.1.0)
     thias-squid3 (1.0.2)
       puppetlabs-stdlib (>= 2.4.0)
 
-GIT
-  remote: https://github.com/Katello/puppet-katello_devel
-  ref: master
-  sha: ac9228481d8d76b87e655e76f5fae317330dfaf4
-  specs:
-    katello-katello_devel (1.0.0)
-      katello-candlepin (< 5.0.0, >= 4.0.0)
-      katello-certs (< 4.0.0, >= 2.0.0)
-      katello-katello (< 8.0.0, >= 6.1.0)
-      katello-pulp (< 6.0.0, >= 4.3.0)
-      katello-qpid (< 4.0.0, >= 3.0.0)
-      puppet-extlib (< 3.0.0, >= 0.10.4)
-      puppetlabs-apache (< 3.0.0, >= 1.0.0)
-      puppetlabs-postgresql (< 6.0.0, >= 3.0.0)
-      puppetlabs-stdlib (< 5.0.0, >= 4.0.0)
-      puppetlabs-vcsrepo (< 3.0.0, >= 1.0.0)
-
 DEPENDENCIES
-  katello-candlepin (< 4.1.0, >= 4.0.0)
-  katello-certs (< 3.1.0, >= 3.0.0)
+  katello-candlepin (< 5.1.0, >= 5.0.0)
+  katello-certs (< 3.0.2, >= 3.0.1)
   katello-common (< 1.1.0, >= 1.0.1)
-  katello-foreman_proxy_content (< 6.1.0, >= 6.0.0)
-  katello-katello (< 7.1.0, >= 7.0.0)
-  katello-katello_devel (>= 0)
-  katello-pulp (< 5.1.0, >= 5.0.0)
-  katello-qpid (< 3.2.0, >= 3.1.0)
+  katello-foreman_proxy_content (< 7.1.0, >= 7.0.0)
+  katello-katello (< 7.3.0, >= 7.2.0)
+  katello-pulp (< 5.4.0, >= 5.3.0)
+  katello-qpid (< 4.0.1, >= 4.0.0)
   katello-service_wait (< 1.1.0, >= 1.0.3)
-  puppetlabs-mysql (< 4.1.0, >= 4.0.0)
```

Comment:
You should run the update with the foreman-installer (1.16) Puppetfile appended to this Puppetfile so these are locked to their versions. Now you're creating a mismatch in versions.